src/aioswitcher/device/tools.py

```diff
@@ -239,13 +243,19 @@ def get_shutter_discovery_packet_index(
     Used in retriving the shutter position/direction from the packet
     (based of device type and circuit number).
     """
-    if circuit_number != 0:
+    if (device_type != DeviceType.RUNNER_S12 and circuit_number != 0) or (
+        device_type == DeviceType.RUNNER_S12
+        and circuit_number != 0
+        and circuit_number != 1
+    ):
```

Comment:
Try to improve readability for this guard (similar to my comments bellow)

Reply:
It is done.